app/Console/AbstractRemoveLabel.php

```diff
@@ -5,32 +5,37 @@
 namespace CultuurNet\UDB3\Silex\Console;
 
 use Broadway\CommandHandling\CommandBus;
+use CultuurNet\UDB3\Label\ReadModels\JSON\Repository\Doctrine\DBALReadRepository;
+use CultuurNet\UDB3\Label\ReadModels\JSON\Repository\InMemoryExcludedLabelsRepository;
+use CultuurNet\UDB3\Label\ReadModels\JSON\Repository\ReadRepositoryInterface;
+use CultuurNet\UDB3\Model\ValueObject\Identity\UUID;
+use CultuurNet\UDB3\Silex\Role\UserPermissionsServiceProvider;
+use CultuurNet\UDB3\StringLiteral;
 use Doctrine\DBAL\Connection;
 use Symfony\Component\Console\Command\Command;
 
 abstract class AbstractRemoveLabel extends Command
 {
-    private Connection $connection;
+    private ReadRepositoryInterface $readRepository;
 
     protected CommandBus $commandBus;
 
     public function __construct(Connection $connection, CommandBus $commandBus)
     {
-        $this->connection = $connection;
+        $this->readRepository = new DBALReadRepository(
+            $connection,
+            new StringLiteral('labels_json'),
+            new StringLiteral('label_roles'),
+            new StringLiteral(UserPermissionsServiceProvider::USER_ROLES_TABLE),
+            new InMemoryExcludedLabelsRepository([])
+        );
         $this->commandBus = $commandBus;
         parent::__construct();
     }
 
-    /*
-     * @return string|false
-     */
-    protected function getLabel(string $labelId)
+    protected function getLabel(string $labelId): string
     {
-        return $this->connection->createQueryBuilder()
-            ->select('unique_col')
-            ->from('labels_unique')
-            ->where('uuid_col = "' . $labelId . '"')
-            ->execute()
-            ->fetchColumn();
+        $uuid = new UUID($labelId);
+        return $this->readRepository->getByUuid($uuid)->getName()->toNative();
```

Comment:
This will crash now if there is no label with the given UUID
```
vagrant@udb3:/var/www/udb-silex$ ./bin/udb3.php offer:remove-label a3e8561e-2903-45fb-891c-a60d3f79fca6 936d0664-566f-4444-a284-0d09f4057cbf
PHP Fatal error:  Uncaught Error: Call to a member function getName() on null in /var/www/udb-silex/app/Console/AbstractRemoveLabel.php:39
Stack trace:
#0 /var/www/udb-silex/app/Console/RemoveLabelOffer.php(28): CultuurNet\UDB3\Silex\Console\AbstractRemoveLabel->getLabel()
#1 /var/www/udb-silex/vendor/symfony/console/Command/Command.php(255): CultuurNet\UDB3\Silex\Console\RemoveLabelOffer->execute()
#2 /var/www/udb-silex/vendor/symfony/console/Application.php(992): Symfony\Component\Console\Command\Command->run()
#3 /var/www/udb-silex/vendor/symfony/console/Application.php(255): Symfony\Component\Console\Application->doRunCommand()
#4 /var/www/udb-silex/vendor/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun()
#5 /var/www/udb-silex/bin/udb3.php(163): Symfony\Component\Console\Application->run()
#6 {main}
  thrown in /var/www/udb-silex/app/Console/AbstractRemoveLabel.php on line 39
```

Reply:
Correct I was a bit too harsh when I tried to minimize code.